packages/compass-saved-aggregations-queries/src/stores/open-item.ts

```diff
@@ -112,91 +125,188 @@ export type Actions =
+  }
+
+  if (
+    isAction<ConnectionSelectedAction>(action, ActionTypes.ConnectionSelected)
+  ) {
+    return {
+      ...state,
+      selectedConnection: action.connection,
+      selectedDatabase: null,
+      databases: [],
+      databasesStatus: 'initial',
+      collections: [],
+      collectionsStatus: 'initial',
+      selectedCollection: null,
+    };
+  }
+
+  if (isAction<SelectDatabaseAction>(action, ActionTypes.SelectDatabase)) {
+    return {
+      ...state,
+      selectedDatabase: action.database,
+      collections: [],
+      collectionsStatus: 'initial',
+      selectedCollection: null,
+    };
+  }
+
+  if (isAction<LoadDatabasesAction>(action, ActionTypes.LoadDatabases)) {
+    return {
+      ...state,
+      databasesStatus: 'fetching',
+    };
+  }
+
+  if (
+    isAction<LoadDatabasesErrorAction>(action, ActionTypes.LoadDatabasesError)
+  ) {
+    return {
+      ...state,
+      databasesStatus: 'error',
+    };
+  }
+
+  if (
+    isAction<LoadDatabasesSuccessAction>(
+      action,
+      ActionTypes.LoadDatabasesSuccess
+    )
+  ) {
+    return {
+      ...state,
+      databases: action.databases,
+      databasesStatus: 'ready',
+    };
+  }
+
+  if (isAction<SelectCollectionAction>(action, ActionTypes.SelectCollection)) {
+    return {
+      ...state,
+      selectedCollection: action.collection,
+    };
+  }
+
+  if (isAction<LoadCollectionsAction>(action, ActionTypes.LoadCollections)) {
+    return {
+      ...state,
+      collectionsStatus: 'fetching',
+    };
+  }
+
+  if (
+    isAction<LoadCollectionsErrorAction>(
+      action,
+      ActionTypes.LoadCollectionsError
+    )
+  ) {
+    return {
+      ...state,
+      collectionsStatus: 'error',
+    };
+  }
+
+  if (
+    isAction<LoadCollectionsSuccessAction>(
+      action,
+      ActionTypes.LoadCollectionsSuccess
+    )
+  ) {
+    return {
+      ...state,
+      collections: action.collections,
+      collectionsStatus: 'ready',
+    };
+  }
+
+  if (
+    isAction<UpdateNamespaceCheckedAction>(
+      action,
+      ActionTypes.UpdateNamespaceChecked
+    )
+  ) {
+    return {
+      ...state,
+      updateItemNamespace: action.updateItemNamespace,
+    };
+  }
+
+  return state;
+};
+
+const getDataServiceAndInstanceForConnection =
+  (
+    connection: string | null
+  ): SavedQueryAggregationThunkAction<
+    | {
+        dataService?: never;
+        instance?: never;
+        error: Error;
+      }
+    | {
+        dataService: DataService;
+        instance: MongoDBInstance;
+        error?: never;
+      }
+  > =>
+  (_dispatch, _getState, { connectionsManager, instancesManager }) => {
+    if (!connection) {
       return {
-        ...state,
-        collectionsStatus: 'error',
+        error: new Error('Connection not provided'),
       };
-    case ActionTypes.LoadCollectionsSuccess:
+    }
+
+    const dataService =
+      connectionsManager.getDataServiceForConnection(connection);
+    if (!dataService) {
```

Comment:
I asked in another PR and I feel like it's worth asking again: do we ever expect in the code not to get data service back and handle it by not throwing an error when using manager in actions? Should this logic be part of the method if we always throw if data service is not returned? Same for instance model

Reply:
Yea I am literally doing this `if not throw` everywhere, will definitely make this part of the getDataService interface. Thank you 👍